src/LibraryManager.Contracts/FileHelpers.cs

```diff
@@ -17,6 +17,42 @@ namespace Microsoft.Web.LibraryManager.Contracts
     /// </summary>
     public static class FileHelpers
     {
+        /// <summary>
+        /// Writes the stream to a temporary file first, then moves the temporary file to the destination file
+        /// </summary>
+        /// <param name="fileName"></param>
+        /// <param name="sourceStream"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        public static async Task<bool> SafeWriteToFileAsync(string fileName, Stream sourceStream, CancellationToken cancellationToken)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            string tempFileName = Path.Combine(
+                Environment.GetFolderPath(Environment.SpecialFolder.InternetCache),
```

Comment:
>Environment.SpecialFolder.InternetCache [](start = 42, length = 39)

What would this be on a non-Windows box?

Reply:
Not sure, but I would hope it would be a meaningful value? Do we have a linux box handy? :)

Reply:
Tried this on Ubuntu, `Environment.SpecialFolder.InternetCache` just returns the string `InternetCache`.

`GetFolderPath(Environment.SpecialFolder.InternetCache)` returns an empty string.

Reply:
I guess there is no guarnatee that Path.GetTempFileName will return correct results on all platforms either, so I added a fallback to the old behavior.